src/databricks/labs/remorph/snow/sql_transpiler.py

```diff
@@ -45,6 +65,58 @@ def parse_sql_content(self, sql, file_name):
                     for select in expr.find_all(exp.Select, exp.Join, exp.With, bfs=False):
                         yield self._find_root_tables(select), child
 
+    @staticmethod
+    def safe_parse(statements: str, read: Dialect) -> tuple[list[exp.Expression], list[str]]:
+        dialect = read
+        errors = []
+        try:
+            tokens = dialect.tokenize(sql=statements)
+        except TokenError as e:
+            error_statement = format_error_message("TOKEN ERROR", e, statements)
+            errors.append(error_statement)
+            return [], errors
+
+        total = len(tokens)
+        chunks: list[list[Token]] = [[]]
+        original_sql_chunks: list[str] = []
+        current_sql_chunk = []
+
+        # Split tokens into chunks based on semicolons(or other separators)
+        # Need to define the separator in Class Tokenizer
+        for i, token in enumerate(tokens):
+            current_sql_chunk.append(token.text)
+            if token.token_type in {TokenType.SEMICOLON}:
```

Comment:
We Should check if the are followed by space or "\n" as well? to be safe.

Reply:
```suggestion
           keyword_dict = { TokenType.COMMAND, TokenType.CREATE, TokenType.ALTER, TokenType.GRANT, TokenType.INSERT, TokenType.With} 
```
Can you create a dictionary and check the next token is not in this list. it is not exhaustive, but we can keep adding if we find bugs.

Reply:
As discussed this will not be required, just add negative test case to eliminate this ambiguity.